### PR TITLE
raftstore-v2: clean up import sst file only if the flushed index is greater than the applied index 

### LINKED ISSUE #15018: [Dynamic Region] the available size is a big gap between tikvs and leading to the capacity of one tikv is insufficient
### What version of TiKV are you using?
./tikv-server -V
 TiKV 
Release Version:   7.3.0-alpha
Edition:           Community
Git Commit Hash:   de3d6655c3bb0eceac068207fe7ec8be30bdbcc7
Git Commit Branch: heads/refs/tags/v7.3.0-alpha
UTC Build Time:    2023-06-26 11:13:06
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure
Profile:           dist_release
2023-06-27T08:30:36.095+0800	INFO	k8s/client.go:132	it should be noted that a long-running command will not be interrupted even the use case has ended. For more information, please refer to https://github.com/pingcap/test-infra/discussions/129

### What operating system and CPU are you using?
8c/32g

### Steps to reproduce
1、run tpcc
2、inject network_partition fault between one tikv and others tikv
3、after 10min，recover the fault

### What did you expect?
the available size between tikvs is balanced

### What did happened?
the available size is a big gap between tikvs and leading to the capacity of one tikv is insufficient 
bc9863ae-668b-4f00-b522-3d572fd780ad
d848a265-715b-49dc-a803-ced39bcdbeef
2169c5ea-34bc-447b-87a4-ea64e51a388f
img_v2_1f6911c9-183d-47af-8eac-f4a15304087g

### PR DESCRIPTION
**Pull request by bufferflies**
### What is changed and how it works?

Issue Number: Close #15018

What's Changed:

```commit-message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

Side effects

### Release note 

```release-note
clean up import sst only if the flushed index is greater than the applied index 
```

### CODE REVIEW

File: tests/failpoints/cases/test_import_service.rs [hunk 1]
```diff
@@ -279,45 +279,84 @@ fn test_ingest_sst_v2() {
...
+        let sst_dir = path.path().join("import-sst");
+        for entry in std::fs::read_dir(sst_dir).unwrap() {
+            let entry = entry.unwrap();
+
+            if entry
+                .path()
+                .file_name()
+                .and_then(|n| n.to_str())
+                .unwrap()
+                .contains("0_0_0")
```
Comment (tabokie) [inline tests/failpoints/cases/test_import_service.rs:352]: what is this?

File: tests/failpoints/cases/test_import_service.rs [hunk 2]
```diff
@@ -279,45 +279,84 @@ fn test_ingest_sst_v2() {
...
+    fail::cfg_callback("on_flush_completed", move || {
+        tx.lock().unwrap().send(()).unwrap();
+    })
+    .unwrap();
+    for i in 0..count {
+        cluster.must_put(format!("k-{}", i).as_bytes(), b"v");
+    }
+    cluster.flush_data();
+
+    rx.recv_timeout(std::time::Duration::from_secs(20)).unwrap();
```
Comment (tabokie) [inline tests/failpoints/cases/test_import_service.rs:333]: why wait so long? `flush_data` waits for flush to finish.

File: components/raftstore-v2/src/operation/command/write/ingest.rs [hunk 3]
```diff
@@ -65,36 +65,21 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
...
         &mut self,
         ctx: &mut StoreContext<EK, ER, T>,
-        ssts: Box<[SstMeta]>,
+        ssts: Box<[(SstMeta, u64)]>,
     ) {
         let mut stale_ssts = Vec::from(ssts);
-        let epoch = self.region().get_region_epoch();
-        stale_ssts.retain(|sst| {
-            fail::fail_point!("on_cleanup_import_sst", |_| true);
-            util::is_epoch_stale(sst.get_region_epoch(), epoch)
```
Comment (tabokie) [inline components/raftstore-v2/src/operation/command/write/ingest.rs:74]: what about this check?

File: tests/failpoints/cases/test_import_service.rs (hunk 1 above)
Comment (bufferflies) [reply tests/failpoints/cases/test_import_service.rs:352]: In this begin, import wiil  create one file named as `0_0_0_0.sst`

File: tests/failpoints/cases/test_import_service.rs (hunk 2 above)
Comment (bufferflies) [reply tests/failpoints/cases/test_import_service.rs:333]: yes, I will set it as 100ms .

**Comment by bufferflies** [comment]
close by #15064

**Pull request closed without merging**